pkg/reconciler/autoscaling/resources/target.go

```diff
@@ -23,11 +23,11 @@ import (
 	"knative.dev/serving/pkg/autoscaler"
 )
 
-// ResolveConcurrency takes concurrency knobs from multiple locations and resolves them
-// to the final value to be used by the autoscaler.
-// `target` is the target concurrency that we autoscaler will aim for;
-// `total` is the maximum possible concurrency that is permitted on the pod.
-func ResolveConcurrency(pa *v1alpha1.PodAutoscaler, config *autoscaler.Config) (target float64, total float64) {
+// ResolveMetricTarget takes scaling metric knobs from multiple locations
+// and resolves them to the final value to be used by the autoscaler.
+// `target` is the target value of scaling metric that we autoscaler will aim for;
+// `total` is the maximum possible value of scaling metric that is permitted on the pod.
+func ResolveMetricTarget(pa *v1alpha1.PodAutoscaler, config *autoscaler.Config) (target float64, total float64) {
 	total = float64(pa.Spec.ContainerConcurrency)
```

Comment:
As Victor mentioned, this feels wrong now. Does it need to be sensitive to the Target annotation to pick the right defaults from config?

Reply:
Yes it needs. However currently we only support concurrency.
I didn't change the logic here or introducing new fields since this PR is just about renaming.

What we currently have for scaling metrics and targets are:
1. Annotations:
```
autoscaling.knative.dev/target
autoscaling.knative.dev/targetUtilizationPercentage
autoscaling.knative.dev/targetBurstCapacity
```

They can be reused for other metrics.

2. Configurations in config-autoscaler:
```
target-burst-capacity
container-concurrency-target-percentage
container-concurrency-target-default
```

The first one can be reused for other metric. The later two can not.

3. Spec in Revision:
```
Spec.ContainerConcurrency
```

This is concurrency only.


My plan is to reuse all the three existing annotations and introduce two more field in configmap:
```
target-percentage
requests-per-second-target-default
```
Use these 5 knobs to calculate target value of RPS(BTW I want to rename OPS to RPS as well). So there will be logic change here after the new fields are introduced.

Does it look good to you?

Reply:
Added a TODO as well.